src/lib.rs

```diff
@@ -243,6 +246,25 @@ fn get_num_cpus() -> usize {
     cpus as usize
 }
 
+
+#[cfg(target_os = "macos")]
+fn get_num_physical_cpus() -> usize {
+    let mut cpus: i32 = 0;
+    let mut cpus_size = std::mem::size_of_val(&cpus);
+
+    let sysctl_name = CString::new("hw.physicalcpu").unwrap();
```

Comment:
Instead of an unsafe transmute, there is https://doc.rust-lang.org/stable/std/ffi/struct.CStr.html#method.from_bytes_with_nul

I'd probably also change the `unwrap` to an `expect("there are no nul bytes")`.